qiskit_nature/properties/second_quantization/electronic/dipole_moment.py

```diff
@@ -42,38 +43,70 @@ class DipoleMoment(_IntegralProperty):
     def __init__(
         self,
         axis: str,
-        basis: ElectronicBasis,
-        electronic_integrals: Dict[int, ElectronicIntegrals],
-        shift: Optional[Dict[str, float]] = None,
+        electronic_integrals: List[ElectronicIntegrals],
+        shift: Optional[Dict[str, complex]] = None,
     ):
         """
         Args:
             axis: the name of the Cartesian axis.
-            dipole: an _IntegralProperty property representing the dipole moment operator.
+            dipole: an IntegralProperty property representing the dipole moment operator.
         """
-        super().__init__(self.__class__.__name__, basis, electronic_integrals, shift=shift)
         self._axis = axis
+        name = self.__class__.__name__ + axis.upper()
+        super().__init__(name, electronic_integrals, shift=shift)
+
+    def matrix_operator(self, density: OneBodyElectronicIntegrals) -> OneBodyElectronicIntegrals:
+        """Constructs the operator of this property in matrix-format for a given density.
+
+        An IntegralProperty typically represents an observable which can be expressed in terms of a
+        matrix-formatted operator at a given electronic density. This method must be implemented by
+        a subclass to provide this functionality.
+
+        Args:
+            density: the electronic density at which to compute the matrix operator.
+
+        Returns:
+            OneBodyElectronicIntegrals: the matrix-formatted operator stored as ElectronicIntegrals.
+
+        Raises:
+            NotImplementedError: if no AO-integrals are stored within `self`.
+        """
+        if ElectronicBasis.AO not in self._electronic_integrals.keys():
+            raise NotImplementedError()
 
+        return cast(OneBodyElectronicIntegrals, self.get_electronic_integral(ElectronicBasis.AO, 1))
 
-class TotalDipoleMoment(SecondQuantizedProperty):
+    def interpret(self, result: EigenstateResult) -> None:
+        """Interprets an :class:~qiskit_nature.result.EigenstateResult in this property's context.
+
+        Args:
+            result: the result to add meaning to.
+        """
+        pass
+
+
+class TotalDipoleMoment(CompositeProperty[DipoleMoment], SecondQuantizedProperty):
```

Comment:
With regards to these comments by @woodsp-ibm:
    - https://github.com/Qiskit/qiskit-nature/pull/220#discussion_r654629373
    - https://github.com/Qiskit/qiskit-nature/pull/220#discussion_r654654309

I suggest to rename this property to `ElectronicDipoleMoment`. The `nuclear` parts of the total energy and total dipole moment still (in my opinion) make sense to be stored as part of these properties, but I agree that they should not be part of the `shift` dictionaries. Instead, I would introduce explicit properties:
```python
ElectronicEnergy.nuclear_repulsion_energy
ElectronicDipoleMoment.nuclear_dipole_moment
```
Since these are only used to provide additional context during the `interpret` methods, I think this is a sufficiently simple solution.

Reply:
@woodsp-ibm What about this point? Are you fine with including the `nuclear` fields as "bonus"-information on an otherwise `electronic` property for the sake of simplifying the implementation of the `interpret` method?

Reply:
I know there has been confusion in the past where people had not understood what we computed was just the electronic part. So in the interpret not only do we adjust for any electronic energy parts that get removed classically ahead of the computation but its going to add in the nuclear part too for the total energy - so at that point its no longer the electronic energy. I know we want the total at some point and I am not sure where else this nuclear part could be added in. Maybe again its down to documentation to explain this aspect too.